File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function (headA, headB) {



    let lengthL1 = 0;
    let lengthL2 = 0;
    let currentA = headA;
    let currentB = headB;
    while (currentA) {
        lengthL1++;
        currentA = currentA.next;
    }
    while (currentB) {
        lengthL2++;
        currentB = currentB.next;
    }

    let difference = lengthL1 - lengthL2;
    currentA = headA;
    currentB = headB;
    console.log(difference)
    if (difference < 0) {
        for (let i = 0; i < Math.abs(difference); i++) {
            currentB = currentB.next;
        }
    }
    else {
        for (let i = 0; i <  Math.abs(difference); i++) {
            currentA = currentA.next;
        }
    }

    while (currentA != currentB) {
        currentA = currentA.next
        currentB = currentB.next
    }
    return currentA;






};

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode* getIntersectionNode(ListNode* headA, ListNode* headB) {

        ListNode* currentA = headA;
        ListNode* currentB = headB;
        while (currentA && currentB) {
            if (currentA == currentB) {
                return currentA;
            }
            currentA = currentA->next;
            currentB = currentB->next;
             if (currentA == NULL && currentB == NULL) {
                return NULL;
            }
            
            if (currentA == NULL) {
                currentA = headB;
            }
            if (currentB == NULL) {
                currentB = headA;
            }
        }

        return NULL;
    }
};